sourcetools/buildj9tools.mk

```diff
@@ -30,13 +30,16 @@
 #   DEST_DIR: the location to copy the J9 Java tools generated binaries
 ################################################################################
 
+include ../../spec.gmk
+
```

Comment:
This is making an assumption on position and existence of a spec.gmk which I don't think is the case for all uses, for example a compilation where openjdk extensions do not exist.

Reply:
That spec.gmk exists inside the build/linuxetc directory, and that file definitely exists on a Hotspot build I have. It exists in that directory, with the variable in it.

Note that buildj9tools.mk is copied two layers deeper than spec.gmk (build/linuxetc/vm/sourcetools), so the ../../spec.gmk should work, and I'm not aware of a Java where that is not the case.

Reply:
It might be reasonable to make that include conditional:
```
ifdef SPEC
include $(SPEC)
endif
```

Reply:
Sure. We should put in a default for VERSION_SPECIFICATION if we can't include spec.gmk. If you plan to build java 8 with openj9, I recommend setting the default VERSION_SPECIFICATION value to 7.

Reply:
Why could we not use the default -source and -target values for either java7 or java8 sdk used to compile?  Do this cause some of of problem?  I don't see why spec level details cannot be removed and use the JAVA_BIN passed from where its called from ( runtime/buildtools.mk ).

Reply:
As it is currently written, `VERSION_SPECIFICATION` defaults to 7.
The version requirement is dictated by the source files we need to compile (which was recently downgraded from 8 to 7). I don't see why we can't just always set `-source` and `-target` to 7.

Reply:
Keith is correct in that PREV_JAVA will currently default to 7.

Also, as for the conditional, gnu.org says:

If you want make to simply ignore a makefile which does not exist or cannot be remade, with no error message, use the -include directive instead of include, like this:

\-include filenames…